contracts/ApplicationRegistry.sol

```diff
@@ -196,6 +197,37 @@ contract ApplicationRegistry is Ownable, Pausable, IApplicationRegistry {
         );
     }
 
+    /**
+     * @notice Mark application as complete
+     * @param _applicationId target applicationId which needs to be marked as complete
+     * @param _workspaceId workspace id of application's grant
+     * @param _reasonMetadataHash metadata file hash with application overall feedback
+     */
+    function completeApplication(
+        uint96 _applicationId,
+        uint96 _workspaceId,
+        string memory _reasonMetadataHash
+    ) external whenNotPaused onlyWorkspaceAdmin(_workspaceId) {
+        Application storage application = applications[_applicationId];
+        for (uint48 i = 0; i < application.milestoneCount; i++) {
```

Comment:
Instead of checking if all milestones are done in this function in a loop, it would be better if you check in `approveMilestone` function if the milestone being approved is last or not.
If it is then mark all milestones as done in a separate state variable called something like `milestonesDone`.

Reply:
Thanks for pointing out. Made the changes